01-data-structures/05-hashes-part-2/open_addressing/open_addressing.rb

```diff
@@ -1,30 +1,109 @@
 require_relative 'node'
 
 class OpenAddressing
-  def initialize(size)
+  attr_reader :item_count
+
+  def initialize(size_of)
+    @items = Array.new(size_of)
+    @item_count = 0.0
   end
 
+
+ #Assign an element to the "hash array" given a key and value
   def []=(key, value)
+    arr_index = index(key, size)
+    if @items[arr_index] == nil
+        @items[arr_index] = Node.new(key, value)
+        @item_count += 1
+    elsif (@items[arr_index].key == key) &&
+          (@items[arr_index].value == value)
+      return @items
+    else
+      nextIndex = next_open_index(arr_index)
+      if (@items[arr_index].key == key) &&
+         (@items[arr_index].value != value) &&
+        (nextIndex == -1)
+          resize
+          @items[nextIndex] = Node.new(key, value)
+          @item_count += 1
```

Comment:
The `nextIndex` value could become invalidated after the resize. You should recalculate the index using `index()` again to ensure accuracy.